core/src/main/java/dev/streamx/cli/ingestion/BasePublishUnpublishCommand.java

```diff
@@ -0,0 +1,74 @@
+package dev.streamx.cli.ingestion;
+
+import dev.streamx.cli.exception.IngestionClientException;
+import dev.streamx.clients.ingestion.StreamxClient;
+import dev.streamx.clients.ingestion.exceptions.StreamxClientException;
+import jakarta.inject.Inject;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Spec;
+
+public abstract class BasePublishUnpublishCommand implements Runnable {
+
+  @ArgGroup(exclusive = false)
+  protected IngestionArguments ingestionArguments;
+
+  @Spec
+  protected CommandSpec spec;
+
+  @Inject
+  protected SchemaProvider schemaProvider;
+
+  @Inject
+  protected StreamxClientProvider streamxClientProvider;
+
+  protected abstract void performCommand(StreamxClient client) throws StreamxClientException;
+
+  @Override
+  public void run() {
+    StreamxClient client = createClient();
```

Comment:
Do we need to create client on each command execution ? Cant we reuse client @jgardo-ds wdyt ?

Reply:
Currently the client is (Auto)Closeable, so it probably shouldn't be stored in opened state as a class field 💭 

What its close method does now, is basically closing its `httpClient` instance

Reply:
> Do we need to create client on each command execution ? Cant we reuse client @jgardo-ds wdyt ?

For now it's meaningless as after single command execution whole StraemxCli is shut downed.
We do not have support for batch operations. When we will have then we should refactor it.